File: Lista2/ex4/src/dados/Aluno.java
package dados;

public class Aluno{
    private String nome;
    private int idade;
    private double[] notas = new double[5];
    private int i = 0;

    public String getNome() {
        return this.nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getIdade() {
        return this.idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    public double[] getNotas() {
        return this.notas;
    }

    public void setNota(double nota) {
        if(i > 4)
            return;
        this.notas[i] = nota;
        i += 1;
    }

    public double calcularMedia(){
        double soma = 0;
        for(int i = 0; i < this.notas.length; i += 1){
            soma += notas[i];
        }
        return soma/this.notas.length;
    }

    public String toString(){
        return "Nome: "+this.nome+"/ Idade: "+this.idade+"/ Média: "+calcularMedia();
    }


}

File: Lista3/ex4/dados/Aluno.py
from dados.Pessoa import Pessoa
class Aluno(Pessoa):
    def __init__(self):
        Pessoa.__init__(self)
        self.__notas = []
    
    def setNotas(self, notas: list):
        self.__notas = notas
    
    def getNotas(self):
        return self.__notas
    
    def calcularMedia(self):
        soma = 0
        for i in self.__notas:
            soma += i
        return soma/len(self.__notas)
    
    def __repr__(self):
        string = "Nome: {} | ".format(self.nome) + "Media: {} | ".format(self.calcularMedia()) + "Notas: "
        size = len(self.__notas)
        for i in self.__notas:
            size -= 1
            if(size == 0):
                string += "{} ".format(i)    
            else:
                string += "{}, ".format(i)
        if(self.calcularMedia() >= 7):
            string += "| Status: Aprovado"
        else:
            string += "| Status: Reprovado"
        return string